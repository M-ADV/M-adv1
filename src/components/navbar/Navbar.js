import React from "react";
///import { useDispatch, useSelector } from "react-redux";
import { Link } from "react-router-dom";
import "../../App.css";
///import { logoutAction } from "../../redux/actions/auth";

const Navbar = () => {
  ///const navigate = useNavigate();

  ///const dispatch = useDispatch();

  ///const isAuthenticated = useSelector((s) => s.state.isAuthenticated);
  // const handleLogInOut = () => {
  //   const userId = localStorage.getItem("id");
  //   if (userId) {
  //     localStorage.removeItem("id");
  //     dispatch(logoutAction(navigate));
  //     navigate("/");
  //   } else {
  //     navigate("/login");
  //   }
  // };


  return (
    <div>
      <nav className="navbar navbar-expand-lg navbar-light fixed-top">
        <div className="container-fluid">
          <Link className="navbar-brand text-light" to="/">
            <img
              src="imagi.ico"
              aria-label="favicon"
              width="30"
              height="30"
              className="d-inline-block align-top"
              alt=""
            />
            <b>-Advisor</b>
          </Link>
          <button
            className="navbar-toggler"
            type="button"
            data-bs-toggle="collapse"
            data-bs-target="#navbarNavDropdown"
            aria-controls="navbarNavDropdown"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon"></span>
          </button>
          <div className="collapse navbar-collapse" id="navbarNavDropdown">
            <ul className="navbar-nav ms-auto text-center">
              <li className="nav-item">
                <Link className="nav-item nav-link text-light" to="/">
                  Home
                </Link>
              </li>
              <li className="nav-item">
                <Link className="nav-item nav-link text-light" to="/howtoapply">
                  How to Apply
                </Link>
              </li>
              <li className="nav-item">
                <Link className="nav-item nav-link text-light" to="/services">
                  Services
                </Link>
              </li>
              <li className="nav-item">
                <Link className="nav-item nav-link text-light" to="/about">
                  About
                </Link>
              </li>
              <li className="nav-item">
                <Link className="nav-item nav-link text-light" to="/faqs">
                  FAQS
                </Link>
              </li>


              <li className="nav-item">
                    <Link
                      className="nav-item nav-link text-light"
                      to="/propertydetails"
                    >
                      Property Details
                    </Link>
                  </li>

{/*
                  <li className="nav-item">
                    <Link
                      className="nav-item nav-link text-light"
                      to="/emicheck"
                    >
                      Eligibility Checker
                    </Link>
  </li>*/}
                  <li className="nav-item">
                    <Link
                      className="nav-item nav-link text-light"
                      to="/contactus"
                    >
                      Queries
                    </Link>
                  </li>
              {localStorage.getItem("id") ? (
                <>
                  
                </>
              ) : (
                <>
                 {/*} <li className="nav-item">
                    <Link
                      className="nav-item nav-link text-light"
                      to="/register"
                    >
                      Register
              </Link>
                  </li>*/}
                </>
              )}

             
            </ul>
          </div>
        </div>
      </nav>
    </div>
  );
};

export default Navbar;
