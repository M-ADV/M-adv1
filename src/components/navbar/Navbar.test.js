import { render, screen } from "@testing-library/react";
import { renderComponent } from "../../utils/test_helpers";
import Navbar from "./Navbar";

describe("render Navbar", () => {
  test("render intial navbar page links", () => {
    render(renderComponent(<Navbar />));
    expect(screen.getByRole("link", { name: /-Loan/i })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /Home/i })).toBeInTheDocument();
    expect(
      screen.getByRole("link", { name: /How to Apply/i })
    ).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /Services/i })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /About/i })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /FAQS/i })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /Register/i })).toBeInTheDocument();
  });
  test("render Navbar with Icon", () => {
    render(renderComponent(<Navbar />));
    expect(screen.getByRole("img", { name: /favicon/i })).toBeInTheDocument();
  });

  test("Matching the snapshot of the navbar", async () => {
    const { asFragment } = render(renderComponent(<Navbar />));
    expect(asFragment()).toMatchSnapshot();
  });
});
