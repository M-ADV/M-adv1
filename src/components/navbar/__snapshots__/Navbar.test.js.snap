// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Navbar Matching the snapshot of the navbar 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-lg navbar-light fixed-top"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand text-light"
          href="/"
        >
          <img
            alt=""
            aria-label="favicon"
            class="d-inline-block align-top"
            height="30"
            src="imagi.ico"
            width="30"
          />
          <b>
            -Advisor
          </b>
        </a>
        <button
          aria-controls="navbarNavDropdown"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-bs-target="#navbarNavDropdown"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse"
          id="navbarNavDropdown"
        >
          <ul
            class="navbar-nav ms-auto text-center"
          >
            <li
              class="nav-item"
            >
              <a
                class="nav-item nav-link text-light"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-item nav-link text-light"
                href="/howtoapply"
              >
                How to Apply
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-item nav-link text-light"
                href="/services"
              >
                Services
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-item nav-link text-light"
                href="/about"
              >
                About
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-item nav-link text-light"
                href="/faqs"
              >
                FAQS
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-item nav-link text-light"
                href="/propertydetails"
              >
                Property Details
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-item nav-link text-light"
                href="/contactus"
              >
                Queries
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
